### PR TITLE
ARC-2086: Correct some files (possibly different formats) are not processed correctly in MediaFileChunker

### PR DESCRIPTION
**Pull request by ademgcl**

### CODE REVIEW

File: helper_apps/MediaFileChunker/fileChunker.js [hunk 1]
```diff
@@ -43,18 +48,57 @@ const generateUniqueFolderName = () => {
...
 async function splitMediaFile(inputPath, chunkDurationInSeconds = 600) {
     try {
+        // Create unique folder
+        const uniqueOutputPath = generateUniqueFolderName();
+        fs.mkdirSync(uniqueOutputPath, { recursive: true });
+
+        // Download the file if it's not a local file
+        const isUrl = /^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$/i.test(inputPath);
+        if (isUrl) {
+            inputPath = ensureEncoded(inputPath);
```
Comment (jmacAJ) [inline helper_apps/MediaFileChunker/fileChunker.js:84]: Just curious - why don't we know whether inputPath is encoded at this point?

File: helper_apps/MediaFileChunker/helper.js [hunk 2]
```diff
@@ -28,6 +28,18 @@ async function deleteTempPath(path) {
     }
 }
 
+function ensureEncoded(url) {
+    try {
+        const decodedUrl = decodeURI(url);
+        if (decodedUrl === url) {
```
Comment (jmacAJ) [inline helper_apps/MediaFileChunker/helper.js:34]: Is this part necessary?  decodeURI will just throw if the URI is not encoded right?  So if it doesn't throw you just have a decoded URI which you probably don't need to re-encode?

**Review by jmacAJ — CHANGES_REQUESTED** [review]
Recommend simplifying ensureEncoded

File: helper_apps/MediaFileChunker/helper.js (hunk 2 above)
Comment (ademgcl) [reply helper_apps/MediaFileChunker/helper.js:34]: knuth got me

File: helper_apps/MediaFileChunker/fileChunker.js (hunk 1 above)
Comment (ademgcl) [reply helper_apps/MediaFileChunker/fileChunker.js:84]: we know actually but while testing manually i saw that we could get trapped with this one in direct call so added it

File: helper_apps/MediaFileChunker/helper.js (hunk 2 above)
Comment (ademgcl) [reply helper_apps/MediaFileChunker/helper.js:34]: i think knuth was right after all, it double encodes otherwise

**Pull request merged**